File: app/src/main/java/de/bikebean/app/ui/drawer/log/GithubGistFiles.java
package de.bikebean.app.ui.drawer.log;

import androidx.annotation.NonNull;

class GithubGistFiles {

    private final @NonNull GithubGistTsv sms_tsv;
    private final @NonNull GithubGistTsv state_tsv;
    private final @NonNull GithubGistTsv log_tsv;

    GithubGistFiles(final @NonNull String smsTsv, final @NonNull String stateTsv,
                    final @NonNull String logTsv) {
        this.sms_tsv = new GithubGistTsv(smsTsv);
        this.state_tsv = new GithubGistTsv(stateTsv);
        this.log_tsv = new GithubGistTsv(logTsv);
    }
}

File: app/src/main/java/de/bikebean/app/ui/drawer/log/GithubGistFiles.kt
package de.bikebean.app.ui.drawer.log

import kotlinx.serialization.Serializable

@Serializable
data class GithubGistFiles(
        val sms_tsv: GithubGistTsv,
        val state_tsv: GithubGistTsv,
        val log_tsv: GithubGistTsv
)

internal class GithubGistFilesBuilder(
        smsTsv: String,
        stateTsv: String,
        logTsv: String) {

    private val data = GithubGistFiles(
            GithubGistTsv(smsTsv),
            GithubGistTsv(stateTsv),
            GithubGistTsv(logTsv),
    )

    fun build() = data

}